pkg/offsetguess/offsetguess.go

```diff
@@ -0,0 +1,293 @@
+// +build linux
+
+package offsetguess
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"github.com/iovisor/gobpf/elf"
+
+	"github.com/weaveworks/tcptracer-bpf/pkg/byteorder"
+)
+
+type tcpTracerState uint64
+
+const (
+	uninitialized tcpTracerState = iota
+	checking
+	checked
+	ready
+)
+
+type guessWhat uint64
+
+const (
+	guessSaddr guessWhat = iota
+	guessDaddr
+	guessFamily
+	guessSport
+	guessDport
+	guessNetns
+	guessDaddrIPv6
+)
+
+type tcpTracerStatus struct {
```

Comment:
If it is not officially stated that C and Go has to have the same underlying representation in memory for primitive types (have you checked go-devs mailing list and golang/go github issues?) , then, for the sake of safety, I'm in favor of the CGO version.